boards/shared/Inc/App/App_SharedError.h

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+struct Error;
+struct ErrorList;
+struct ErrorBoardList;
+
+enum ErrorBoard
+{
+    BMS,
+    DCM,
+    DIM,
+    FSM,
+    PDM,
+    NUM_ERROR_BOARDS,
+};
+
+struct ErrorBoardList
```

Comment:
So instead of the original design where the function stores a static array of pointers (e.g. `void ErrorTable_getBoardsWithCriticalErrors(ErrorTable* error_table, ErrorBoard[] boards, int* num_boards), I have the caller provide storage for the pointers. 

I personally prefer this approach a little more. Let me know if you feel strongly about switching about to the static array approach.

Reply:
Nah, I like this as well. :+1: